lib/keygen.hoon

```diff
@@ -29,44 +29,55 @@
   (wallet:dr child-seed)
 ::
 ++  full-wallet-from-ticket
-  |=  [ticket=byts seed-size=@ud sis=(set ship) pass=(unit @t) revs=revisions]
-  =+  owner-seed=seed-size^(argon2u ticket seed-size)
-  (full-wallet-from-seed owner-seed sis pass revs)
-::
-++  full-wallet-from-seed
-  |=  [owner-seed=byts sis=(set ship) pass=(unit @t) revs=revisions]
+  |=  $:  ticket=byts
+          seed-size=@ud
+          sis=(set ship)
+          pass=(unit @t)
+          revs=revisions
+          boot=?
+      ==
+  ^-  vault
+  =/  owner-seed=byts
+    seed-size^(argon2u ticket seed-size)
   =+  dr=~(. sd pass)
-  =+  cn=|=([s=byts m=meta] (child-node-from-seed s m pass))
+  =/  cn
+    |=  m=meta
+    (child-node-from-seed owner-seed m pass)
+  ::
+  :-  `@q`dat.ticket
```

Comment:
Oh yeah, I'll add this feature to ob-js tomorrow.

Reply:
This really should just be that simplified implementation you did with just the star names (though this also supports uneven byte counts, ie `~fun-witbud`).  
ob-js may or may not be the right place for it, but I think it's the best place we have for JS utils relating to urbit name-like types?

Reply:
(See also #824 I guess.)